operator/controllers/common/otel_utils.go

```diff
@@ -0,0 +1,284 @@
+		return err
+	}
+
+	o.TracerProvider = trace.NewTracerProvider(tpOptions...)
+	otel.SetTracerProvider(o.TracerProvider)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+	o.OtelExporter = &otelExporter
+	logger.Info("Successfully initialized OTel collector")
+	return nil
+}
+
+func (o *otelConfig) ShutDown() {
+	if err := o.TracerProvider.Shutdown(context.Background()); err != nil {
+		os.Exit(1)
+	}
+}
+
+func GetOTelTracerProviderOptions(oTelCollectorUrl string) ([]trace.TracerProviderOption, trace.SpanExporter, error) {
+	var tracerProviderOptions []trace.TracerProviderOption
+	var otelExporter trace.SpanExporter
+
+	stdOutExp, err := newStdOutExporter()
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not create stdout OTel exporter: %w", err)
+	}
+	tracerProviderOptions = append(tracerProviderOptions, trace.WithBatcher(stdOutExp))
+
+	if oTelCollectorUrl != "" {
+		// try to set OTel exporter for Jaeger
+		otelExporter, err := newOTelExporter(oTelCollectorUrl)
+		if err != nil {
+			// log the error, but do not break if Jaeger exporter cannot be created
+			logger.Error(err, "Could not set up OTel exporter")
+			return nil, nil, err
+		} else if otelExporter != nil {
+			tracerProviderOptions = append(tracerProviderOptions, trace.WithBatcher(otelExporter))
+		}
+	}
+	tracerProviderOptions = append(tracerProviderOptions, trace.WithResource(newResource()))
+
+	return tracerProviderOptions, otelExporter, nil
+}
+
+func newStdOutExporter() (trace.SpanExporter, error) {
+	return stdouttrace.New(
+		// Use human readable output.
+		stdouttrace.WithPrettyPrint(),
+		// Do not print timestamps for the demo.
+		stdouttrace.WithoutTimestamps(),
+	)
+}
+
+func newOTelExporter(oTelCollectorUrl string) (trace.SpanExporter, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 3*time.Second)
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, oTelCollectorUrl, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create gRPC connection to collector at %s: %w", oTelCollectorUrl, err)
+	}
+	traceExporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithGRPCConn(conn))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create trace exporter: %w", err)
+	}
+	return traceExporter, nil
+}
+
+func newResource() *resource.Resource {
+	r := resource.NewWithAttributes(
+		semconv.SchemaURL,
+		semconv.TelemetrySDKLanguageGo,
+		semconv.ServiceNameKey.String("keptn-lifecycle-operator"),
+		semconv.ServiceVersionKey.String(buildVersion+"-"+gitCommit+"-"+buildTime),
+	)
+	return r
+}
+
+func SetUpKeptnMeters(meter metric.Meter, mgr client.Client) {
+	deploymentActiveGauge, err := meter.AsyncInt64().Gauge("keptn.deployment.active", instrument.WithDescription("a gauge keeping track of the currently active Keptn Deployments"))
+	if err != nil {
+		logger.Error(err, "unable to start OTel")
```

Comment:
let's replace the "unable to start OTel" messages with something more specific about the metric that was attempted to be created, to also fix the Sonar complaint

Reply:
done